src/crush/CrushWrapper.h

```diff
@@ -1094,6 +1094,34 @@ class CrushWrapper {
     for (int i=0; i<numrep; i++)
       out[i] = rawout[i];
   }
+  
+  int check_crush_rule(int ruleset, int type, int size, string &error) {
+    error = "";
+    if (!crush) {
+      error="the crush is not exit ";
+      return -1;
+    }
+    
+    __u32 i;
+    for (i = 0; i < crush->max_rules; i++) {
+      if (crush->rules[i]
+          && crush->rules[i]->mask.ruleset == ruleset
+          && crush->rules[i]->mask.type == type) {
+        if (crush->rules[i]->mask.min_size <= size
+          && crush->rules[i]->mask.max_size >= size) {
+          return i;
```

Comment:
@tchaikov  Thank you ! Sir. Because sometimes the rule no may be diff with the ruleset.So some place we need the rulleno.